apps/cache-service/internal/objectstore/s3.go

```
@@ -350,11 +359,31 @@ func (u *s3Upload) flush(ctx context.Context) error {
+	for i < len(u.pending) && taken < n {
+		part := u.pending[i]
+		want := n - taken
+		if part.size <= want {
+			readers = append(readers, io.LimitReader(part.body, part.size))
+			taken += part.size
+			i++
+			continue
+		}
+		readers = append(readers, io.LimitReader(part.body, want))
+		u.pending[i] = pendingPart{body: part.body, size: part.size - want}
+		taken += want
```

golangci-lint (2.12.2)

[error] ineffectual assignment to taken

(ineffassign)